### PR TITLE
fix(video-conferencing-service): fix session APIs - DB and webhook issues and some refactoring

### PR DESCRIPTION
**Pull request by anitsingh-sf**
## Description

service(video-conferencing-service) fix session APIs - DB and webhook issues and some refactoring

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [x] Performed a self-review of my own code
- [x] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql [hunk 1]
```diff
@@ -16,7 +16,7 @@ CREATE TABLE videochatms.video_session_details
   end_time					timestamptz,
   created_on 				timestamptz DEFAULT current_timestamp NOT NULL ,
   modified_on 			timestamptz ,
-  is_deleted 				bool DEFAULT false NOT NULL ,
+  deleted 				  bool DEFAULT false NOT NULL ,
```
Comment (vineet-suri) [inline services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql:19]: Why this change?

File: services/video-conferencing-service/src/controllers/video-chat-session.controller.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ export class VideoChatSessionController {
     @repository(AuditLogsRepository)
     private readonly auditLogRepository: AuditLogsRepository,
     @repository(SessionAttendeesRepository)
-    private readonly sessionAttendeesRepossitory: SessionAttendeesRepository,
+    private readonly sessionAttendeesRepository: SessionAttendeesRepository,
     @inject(VonageBindings.config)
```
Comment (vineet-suri) [inline services/video-conferencing-service/src/controllers/video-chat-session.controller.ts:37]: Use optional argument here

File: services/video-conferencing-service/src/controllers/video-chat-session.controller.ts [hunk 3]
```diff
@@ -181,13 +182,20 @@ export class VideoChatSessionController {
...
+         the scheduled time`;
         auditLogPayload.after = {errorMessage};
         this.auditLogRepository.create(auditLogPayload);
-        throw new Error(errorMessage);
+        throw new HttpErrors.BadRequest(errorMessage);
       }
     }
 
+    if(!session.startTime) {
+      this.videoChatSessionRepository.updateById(session.id, {
```
Comment (vineet-suri) [inline services/video-conferencing-service/src/controllers/video-chat-session.controller.ts:194]: start time should be decided when the first client logs into session, not based on time of issue of token??

File: services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql (hunk 1 above)
Comment (anitsingh-sf) [reply services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql:19]: VideoChatSession model extends SoftDeleteEntity which has "deleted" property.

File: services/video-conferencing-service/src/controllers/video-chat-session.controller.ts (hunk 2 above)
Comment (anitsingh-sf) [reply services/video-conferencing-service/src/controllers/video-chat-session.controller.ts:37]: for config? if yes, why? because timeToStart comes from config.

File: services/video-conferencing-service/src/controllers/video-chat-session.controller.ts (hunk 3 above)
Comment (anitsingh-sf) [reply services/video-conferencing-service/src/controllers/video-chat-session.controller.ts:194]: its the time of issue of first token for that particular session, only updated once. isnt it same as the first client logs in, token is requested before logging in to the session, right?

File: services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql (hunk 1 above)
Comment (bhavyasf) [reply services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql:19]: @anitsingh-sf  created_by and modified_by must be string, because they send uuid user id, otherwise it will throw NaN error in PSQL.
cc @vineet-suri

File: services/video-conferencing-service/src/controllers/video-chat-session.controller.ts (hunk 2 above)
Comment (vineet-suri) [reply services/video-conferencing-service/src/controllers/video-chat-session.controller.ts:37]: optional flag: "A boolean flag to indicate if the dependency is optional. If it's set to true and the binding is not bound in a context, the resolution will return undefined instead of throwing an error."

File: services/video-conferencing-service/src/controllers/video-chat-session.controller.ts (hunk 3 above)
Comment (vineet-suri) [reply services/video-conferencing-service/src/controllers/video-chat-session.controller.ts:194]: Yes, but that handling is on client side.To fill it via webhooks ( while saving attendees), seems more accurate starting time for meeting.

File: services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql (hunk 1 above)
Comment (bhavyasf) [reply services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql:19]: and @vineet-suri  i think session_id must be unique for prevention of any redundancy of data.

File: services/video-conferencing-service/src/providers/vonage/vonage.service.ts [hunk 4]
```diff
@@ -80,6 +80,7 @@ export class VonageService implements VonageVideoChat {
             mediaMode: sessionCreationOptions.mediaMode,
             archiveMode: sessionCreationOptions.archiveMode,
             sessionId: sessionResponse.sessionId,
+            isArchived: sessionCreationOptions.archiveMode === VonageEnums.ArchiveMode.Always ? true : false
```
Comment (bhavyasf) [inline services/video-conferencing-service/src/providers/vonage/vonage.service.ts:83]:
```suggestion
            isArchived: Boolean(sessionCreationOptions.archiveMode)
```

File: services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql (hunk 1 above)
Comment (vineet-suri) [reply services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql:19]: yes, and meetingLink as well.

Comment (bhavyasf) [reply services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql:19]: also we need to change createdBy and modifiedBy as varchar!

Comment (anitsingh-sf) [reply services/video-conferencing-service/migrations/sqls/20200526081137-db-init-up.sql:19]: done

File: services/video-conferencing-service/src/controllers/video-chat-session.controller.ts (hunk 3 above)
Comment (anitsingh-sf) [reply services/video-conferencing-service/src/controllers/video-chat-session.controller.ts:194]: to fill it via webhook same error will be faced as with SessionAttendeesRepository, we would have to change the VideoChatSessionRepository from DefaultUserModifyCrudRepository to DefaultCrudRepository, because requests via webhooks dont have a current user.

File: services/video-conferencing-service/src/providers/vonage/vonage.service.ts (hunk 4 above)
Comment (anitsingh-sf) [reply services/video-conferencing-service/src/providers/vonage/vonage.service.ts:83]: ok.

File: services/video-conferencing-service/src/controllers/video-chat-session.controller.ts (hunk 2 above)
Comment (anitsingh-sf) [reply services/video-conferencing-service/src/controllers/video-chat-session.controller.ts:37]: done
